File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/factory_method/IOperationFactory.java
package jv.com.tencent.cain.design_pattern.factory_method;


/**
 * @author cainjiang
 * @date 2019-05-20
 */
public interface IOperationFactory {
    /**
     * 创建具体的操作方法实例
     *
     * @return
     */
    public Operation createOpration();

}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/factory_method/IOperationFactory.kt
package kt.com.tencent.cain.desigin_pattern.factory_method

import kt.com.tencent.cain.desigin_pattern.simple_factoty.Operation

/**
 * @author cainjiang
 * @date 2019-05-21
 */
interface IOperationFactory {
    fun createOpration(): Operation
}